### PR TITLE
More script/build improvements (part 1)

### PR DESCRIPTION
**Pull request by Margen67**
Split from #15234

### CODE REVIEW

File: scripts/linux/build-appimage.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
-#!/bin/bash -ex
+#!/usr/bin/env bash
+
+set -ex
 
 # use RAM disk if possible
 if [ "$CI" == "" ] && [ -d /dev/shm ]; then
```
Comment (janisozaur) [inline scripts/linux/build-appimage.sh:6]: the use of `&&` here is unlike in other places, where it's inside the `[`, `]`, any comment on this?

File: scripts/get-discord-rpc [hunk 2]
```diff
@@ -1,4 +1,6 @@
-#!/bin/bash -e
+#!/usr/bin/env bash
+
+set -e
 
 if [[ $(uname) == "Linux" ]]; then
```
Comment (janisozaur) [inline scripts/get-discord-rpc:5]: I noticed you changed exactly same code in `check-code-formatting`, but not here, any reasoning?

File: scripts/linux/build-appimage.sh (hunk 1 above)
Comment (Margen67) [reply scripts/linux/build-appimage.sh:6]: If they behave the same they should be made consistent. (inside the braces)

File: scripts/get-discord-rpc (hunk 2 above)
Comment (Margen67) [reply scripts/get-discord-rpc:5]: This line doesn't need to be changed.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -373,7 +373,7 @@ jobs:
             ./gradlew app:assemblePR
           popd
           mkdir -p artifacts
-          mv src/openrct2-android/app/build/outputs/apk/arm/pr/app-arm-pr.apk artifacts/openrct2-arm.apk
+          cp src/openrct2-android/app/build/outputs/apk/arm/pr/app-arm-pr.apk artifacts/openrct2-arm.apk
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:376]: Why?

File: scripts/linux/build-appimage.sh (hunk 1 above)
Comment (tupaschoal) [reply scripts/linux/build-appimage.sh:6]: The `$CI` line could also use the `-z` treatment done in the second to last commit, right?

**Comment by tupaschoal** [comment]
Beware of adding more commits here, remember it's been split off because the original PR was too long!

File: .github/workflows/ci.yml (hunk 3 above)
Comment (tupaschoal) [reply .github/workflows/ci.yml:376]: Still waiting on the reason for this change, otherwise I think the PR looks fine.

Comment (Margen67) [reply .github/workflows/ci.yml:376]: It's to "preserve" the original file, but I can undo it.

Comment (tupaschoal) [reply .github/workflows/ci.yml:376]: It's already cleaned up soon after this, isn't it? It's moved to `artifacts` which will then be used to upload and then the action finishes. `mv` should technically be faster

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**